app/models/user.rb

```diff
@@ -55,4 +55,12 @@ def requester?
   def possible_projects_professional
     Project.unassigned.joins(skills: :users).where("user_skills.user_id = ?", self.id)
   end
+
+  def role_to_add
+    (Role.signup_roles - roles)
+  end
+
+  def add_role
```

Comment:
Could we name this something like add_extra_role ?  I'm foreseeing some confusing about why we use add_role here but not on the create. If I'm understanding what I'm looking at, it looks like this method is used in combo with role_to_add so it dynamically adds the requester/professional role that they don't already have?